src/services/users.ts

```
@@ -0,0 +1,352 @@
+        .single()
+
+      if (profileError) {
+        // If profile creation fails, we should clean up the auth user
+        // Note: In production, you might want to handle this differently
+        console.error('Profile creation failed, auth user created:', authData.user.id)
+        throw new Error(`Error creating user profile: ${profileError.message}`)
+      }
+
+      return profileData as User
+    } catch (error) {
+      throw error
```

Biome (1.9.4)

[error] The catch clause that only rethrows the original error is useless.

An unnecessary catch clause can be confusing.
Unsafe fix: Remove the try/catch clause.


(lint/complexity/noUselessCatch)